core/iwasm/libraries/debug-engine/debug_engine.c

```diff
@@ -21,11 +21,28 @@ typedef struct WASMDebugEngine {
     int platform_port;
     int process_base_port;
     bh_list debug_instance_list;
+    korp_mutex instance_list_lock;
     bool active;
 } WASMDebugEngine;
 
 static WASMDebugEngine *g_debug_engine;
 
+static uint32 current_instance_id = 1;
+
+static uint32
+allocate_instance_id()
```

Comment:
how about [GCC atomic builtins](https://gcc.gnu.org/onlinedocs/gcc/_005f_005fatomic-Builtins.html)?

Reply:
I‘ve also considered this, but I concern about the compiler compatibility. The GCC atomic builtins are optional features so there may be some compiler who doesn't implement it.

Also, in my opinion, these codes are not on the critical path, the mutex lock should not raise performance issues

@wenyongh how do you think about this?

Reply:
Should not use the optional function which might lead to link error in some platform or compiler. Normally here we should use the API provided by platform layer.